src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -487,11 +488,16 @@ static MvcCoreLoggerExtensions()
                19,
                "Could not bind to model with name '{ModelName}' and type '{ModelType}' as the request did not have a content type of either 'application/x-www-form-urlencoded' or 'multipart/form-data'.");
 
-            _cannotCreateHeaderModelBinder = LoggerMessage.Define<Type>(
+            _cannotCreateHeaderModelBinderCompatVersion_2_0 = LoggerMessage.Define<Type>(
                LogLevel.Debug,
                20,
                "Could not create a binder for type '{ModelType}' as this binder only supports 'System.String' type or a collection of 'System.String'.");
 
+            _cannotCreateHeaderModelBinder = LoggerMessage.Define<Type>(
+               LogLevel.Debug,
+               20,
```

Comment:
Should this have a different event id? 20's already used no?

Reply:
I had commented on this here https://github.com/aspnet/Mvc/pull/7294/files#r164833556 . Let me know if you agree.